src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionUsage.java

```diff
@@ -130,10 +131,26 @@ public final boolean getHasNonDevUseExtension() {
     return getProxies().stream().anyMatch(p -> !p.isDevDependency());
   }
 
+  @GenerateTypeAdapter
+  public record RepoOverride(
+      // The apparent name of the overriding repo in the root module.
+      String overridingRepoName,
+      // Whether this override should apply to an existing repo.
+      boolean mustExist,
+      Location location) {}
+
+  /**
+   * Maps repo names local to the extension to repo they are overridden by.
```

Comment:
for a bit more clarity: "Contains information about overrides that apply to repos generated by this extension. Keyed by the extension-local repo name." (The value type meaning is self evident.)